core/src/peer_id.rs

```diff
@@ -123,7 +123,7 @@ impl PeerId {
     /// equality of peer IDs. That is, two peer IDs may be considered equal
     /// while having a different byte representation as per `into_bytes`.
     pub fn into_bytes(self) -> Vec<u8> {
-        self.multihash.to_vec()
+        self.multihash.as_ref().clone()
     }
```

Comment:
I guess this should be replaced with an `AsRef<[u8]>` or `fn as_bytes(&self) -> &[u8]` and defer cloning to the call site if necessary

Reply:
and an even better solution would be to use `Multihash` internally, since it is stack allocated and use an `Arc<PeerId>` if efficient cloning is required. For cid we just implement `Copy`. This also has the advantage that `Vec<PeerId>` only requires a single allocation.

Reply:
Seems like the reason for this is that the kademlia buckets require something that implements `Borrow<[u8]>`  so I guess this is the best that can be done for now